src/docfx/build/page/BuildPage.cs

```diff
@@ -296,7 +294,8 @@ private static async Task<(List<Error> errors, JObject model)> LoadJson(Context
                 var (deserializeErrors, landingData) = JsonUtility.ToObject<LandingData>(pageModel);
                 errors.AddRange(deserializeErrors);
 
-                var htmlDom = HtmlUtility.LoadHtml(await RazorTemplate.Render(file.Mime, landingData)).PostMarkup(context.Config.DryRun);
+                var razorHtml = RazorTemplate.Render(file.Mime, landingData).GetAwaiter().GetResult();
```

Comment:
seems like `task.GetAwaiter().GetResult()` and `task.Wait()` are essentially the same? Both calling `task.InternalWait`
Check the [call](https://github.com/dotnet/runtime/blob/bedc8d505aa4c42d22914ffb811d7224d871dec7/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs#L2693) within `Task.Wait()`
and the [call](https://github.com/dotnet/runtime/blob/bedc8d505aa4c42d22914ffb811d7224d871dec7/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/TaskAwaiter.cs#L142) within `TaskAwaiter`

Reply:
They will [throw different exceptions](https://stackoverflow.com/questions/17284517/is-task-result-the-same-as-getawaiter-getresult)

Reply:
How about [`task.WaitAndUnwrapException()`](https://stackoverflow.com/questions/9343594/how-to-call-asynchronous-method-from-synchronous-method-in-c?answertab=votes#tab-top) then?

Reply:
That is a third party helper function. The established pattern is to use `GetAwaiter().GetResult()`

Reply:
right